public/pages/workflows/import_workflow/import_workflow_modal.tsx

```diff
@@ -86,6 +87,11 @@ export function ImportWorkflowModal(props: ImportWorkflowModalProps) {
     return description.length > MAX_DESCRIPTION_LENGTH;
   }
 
+  // States for tracking workflow template compatibility with current data source version
+  const [isCompatible, setIsCompatible] = useState<boolean>(true);
+  const [inCompatibleDataSourceVersion, setInCompatibleDataSourceVersion] = useState<string | undefined>();
```

Comment:
similar to other comment, this state var shouldn't be necessary. Can just fetch the data source version after getting the data source ID (under line 64), and if incompatible, include in the callout directly.